main.py

```diff
@@ -478,14 +473,14 @@ def multmod_with_ibdwt(signal1, signal2, prime_exponent,
 
 def rollback():
     if jnp.shape(gec_s_saved) is None:
-        raise Exception(
-            "Gerbicz error checking found an error but had nothing to rollback to. Exiting")
+        msg = "Gerbicz error checking found an error but had nothing to rollback to. Exiting"
+        raise Exception(msg)
     if jnp.shape(gec_d_saved) is None:
-        raise Exception(
-            "Gerbicz error checking found an error but had nothing to rollback to. Exiting")
+        msg = "Gerbicz error checking found an error but had nothing to rollback to. Exiting"
+        raise Exception(msg)
     if gec_i_saved is None:
-        raise Exception(
-            "Gerbicz error checking found an error but had nothing to rollback to. Exiting")
+        msg = "Gerbicz error checking found an error but had nothing to rollback to. Exiting"
+        raise Exception(msg)
```

Comment:
Since it's all the same error, typically I'd just define it at the top of the function or in some class as a variable of errors. The latter example: 
```
GERBICZ_ERROR_GENERAL =  "Gerbicz error checking found an error but had nothing to rollback to. Exiting"
```

Reply:
Good catch! Ruff made these changes automatically, but I did not notice that they were all the same exception. Anyway, thanks, I will make your suggested change.